src/zod-extensions.ts

```diff
@@ -42,6 +44,23 @@ export function extendZodWithOpenApi(zod: typeof z) {
         },
       },
     });
+
+    if (isZodType(this, 'ZodObject')) {
+      const initialExtend = this.extend;
+
+      // TODO: This does an overload everytime. So .extend().openapi() makes this change twice
+      result.extend = function (...args: any) {
```

Comment:
Actually wouldn't this be applied to an object only if it already had `openapi` called? Is that the intention? Why not override it globally and only use it if necessary?

Reply:
Overriding it globabally was the initial idea, yes. The problem there was that if we look at the zod implementation we have the following:
```js
class ZodObject extends ZodType {
  constructor() {
      super(...arguments);
      // some other stuff
      this.extend = AugmentFactory(this._def);
   }
```
This means that we cannot actually modify `zod.ZodObject.prototype.extend` since it is not a "static" (as in already exists on the class) method but is being generated upon object construction.

That being said I still think this approach is elegant since in order to "create a refId" (be it manually or from the registry) you need to call the `.openapi` method. So in some sense we would only modify objects that are being used with our library.